contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -179,6 +179,11 @@ Http::FilterTrailersStatus Filter::encodeTrailers(Http::ResponseTrailerMap& trai
 void Filter::onDestroy() {
   ENVOY_LOG(debug, "golang filter on destroy");
 
+  // do nothing, stream reset may happen before entering this filter.
+  if (req_ == nullptr) {
```

Comment:
There is a potential TODO here, and it is best to enter the go plugin for this scenario in the future.

Reply:
Thanks, we could do it in the feature, and that needs more changes, leave it for now.